docs/pages/getting-started/5-irma-contract.rst

```diff
@@ -0,0 +1,343 @@
+.. _irma-contract:
+
+Signing a contract with IRMA
+############################
+
+This getting started manual shows how to successfully use IRMA to sign a contract.
+Contracts are used within the Nuts ecosystem to identify a user to other network participants.
+It also connects a user to the care organization that user is currently working for.
+The signed contract can be seen as a network enabled session token and should therefore be stored alongside the user session.
+The contract is required for every request that results in personal and/or medical data being retrieved.
+
+Basic requirements
+******************
+
+To use IRMA as a means for signing a contract, the following is required:
+
+- the user has the IRMA app installed on an android or iOS device with camera and an internet connection.
+- the user interacts with the XIS/ECD via a recent browser capable of running javascript.
+- the vendor has a Nuts node running.
+
+IRMA flow
+*********
+
+We use the Nuts node as IRMA server and as tool to start an IRMA session. This follow the flow as described on this `IRMA Github page <https://github.com/privacybydesign/irma-frontend-packages#supported-irma-flows>`_.
+The XIS/ECD will have to provide two endpoints for the frontend. One endpoint to start a session and one to get the session result.
+More info on these endpoints will be provided further down.
+
+
+Configuring the Nuts node
+*************************
+
+In the contract signing flow, the device running the IRMA app communicates with the Nuts node directly.
+Therefore the Nuts node needs to be accessible to the public internet.
+All APIs on the Nuts node starting with ``/public`` must be accessible over https without any security measures in place.
+A domain must also be available which resolves to those APIs.
+The domain must be configured on the Nuts node:
+
+.. code-block:: yaml
+
+  auth:
+    publicurl: https://example.com
+
+The Nuts APIs used for signing will embed this URL in the QR code shown to the user.
+The javascript in the frontend will also use this domain (exposed via the QR code) to check the status of the signing session.
+Therefore the domain which serves the frontend must be able to do requests to that domain.
+The browser will require CORS headers to be configured on the domain configured in the Nuts node config.
+This can be done by the following snippet:
+
+.. code-block:: yaml
+
+    http:
+        default:
+            cors:
+                origin: "other.com"
+
+Where *other.com* is the domain serving the frontend. For development purposes ``*`` is also allowed.
+If the public APIs are mounted on a different port/interface in the nuts config then the ``default`` key should be changed to ``public`` in the example above.
+
+The XIS/ECD communicates with the Nuts node via the ``/internal`` APIs.
+
+Setting up the frontend
```

Comment:
Should we add a link to Demo EHR repo for convenience?

Reply:
maybe later, not when it's incomplete.